packages/eas-cli/src/build/android/applicationId.ts

```diff
@@ -7,17 +7,35 @@ import path from 'path';
 
 import log from '../../log';
 import {
+  ensureAppIdentifierIsDefinedAsync,
   getAndroidApplicationIdAsync,
   getProjectConfigDescription,
 } from '../../project/projectUtils';
 import { promptAsync } from '../../prompts';
 import { gitAddAsync } from '../../utils/git';
+import { Platform } from '../types';
 
 enum ApplicationIdSource {
   AndroidProject,
   AppJson,
 }
 
+function isApplicationIdValid(applicationId: string): boolean {
+  return /^[a-zA-Z][a-zA-Z0-9_]*(\.[a-zA-Z][a-zA-Z0-9_]*)+$/.test(applicationId);
```

Comment:
> Did you find this regex somewhere? 

Yes I explained it in PR description

>  If yes, maybe add a comment with a link to the resource.

I don't think that linking to expo-cli sources would be good idea, it will change in the future for sure